gslab_scons/builders/build_latex.py

```diff
@@ -0,0 +1,60 @@
+import os
+import subprocess
+import shutil
+import gslab_scons.misc as misc
+from gslab_scons import log_timestamp
+from gslab_scons._exception_classes import ExecCallError
+
+def build_latex(target, source, env):
+    '''Compile a pdf from a LaTeX file
+
+    This function is a SCons builder that compiles a .tex file
+    as a pdf and places it at the path specified by target.
+
+    Parameters
+    ----------
+    target: string or list 
+        The target of the SCons command. This should be the path
+        of the pdf that the builder is instructed to compile. 
+    source: string or list
+        The source of the SCons command. This should
+        be the .tex file that the function will compile as a PDF.
+    env: SCons construction environment, see SCons user guide 7.2
+    '''
+
+    # Prelims
+    source      = misc.make_list_if_string(source)
+    target      = misc.make_list_if_string(target)
+
+    source_file = str(source[0])
+    misc.check_code_extension(source_file, '.tex')
+
+    # Set up target file and log file
+    newpdf      = source_file[:-4] + '.pdf'
+    target_file = str(target[0])
+    target_dir  = misc.get_directory(target_file)
+
+    start_time  = misc.current_time()
+
+    try:
+        log_ext = '_%s' % env['log_ext']
+    except KeyError:
+        log_ext = ''
+    log_file = os.path.join(target_dir, ('sconscript%s.log' % log_ext))
+
+    # System call
+    try:
+        command = 'pdflatex -jobname %s %s > %s' % (target_file.replace('.pdf', ''), source_file, log_file)
+        subprocess.check_output(command,
+                                stderr = subprocess.STDOUT,
+                                shell  = True)
+        os.remove(target_file.replace('.pdf', '.log'))
```

Comment:
I think most relevant information is also printed to `sconsript.log`. Below is the comparison for one failed run. Do you think it's still useful to append the `text.log` after `sconscript.log`?

<details> <summary> sconscript.log </summary>
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./source/paper/text.tex
LaTeX2e <2016/03/31>
Babel <3.9r> and hyphenation patterns for 83 language(s) loaded.
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/size12.clo))

! LaTeX Error: Missing \begin{document}.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.5 \author{\noindent Notes}\maketitle
                                      
[1{/usr/local/texlive/2016/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
(./build/paper/text.aux) )
(see the transcript file for additional information)</usr/local/texlive/2016/te
xmf-dist/fonts/type1/public/amsfonts/cm/cmbx12.pfb></usr/local/texlive/2016/tex
mf-dist/fonts/type1/public/amsfonts/cm/cmr12.pfb>
Output written on build/paper/text.pdf (1 page, 18345 bytes).
Transcript written on build/paper/text.log.
 </details>

<details> <summary> text.log </summary>
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016) (preloaded format=pdflatex 2016.5.22)  23 AUG 2017 10:57
entering extended mode
 restricted \write18 enabled.
 %&-line parsing enabled.
**source/paper/text.tex
(./source/paper/text.tex
LaTeX2e <2016/03/31>
Babel <3.9r> and hyphenation patterns for 83 language(s) loaded.
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/size12.clo
File: size12.clo 2014/09/29 v1.4h Standard LaTeX file (size option)
)
\c@part=\count79
\c@section=\count80
\c@subsection=\count81
\c@subsubsection=\count82
\c@paragraph=\count83
\c@subparagraph=\count84
\c@figure=\count85
\c@table=\count86
\abovecaptionskip=\skip41
\belowcaptionskip=\skip42
\bibindent=\dimen102
)

! LaTeX Error: Missing \begin{document}.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.5 \author{\noindent Notes}\maketitle
                                      
You're in trouble here.  Try typing  <return>  to proceed.
If that doesn't work, type  X <return>  to quit.

LaTeX Font Info:    External font `cmex10' loaded for size
(Font)              <14.4> on input line 5.
LaTeX Font Info:    External font `cmex10' loaded for size
(Font)              <7> on input line 5.
[1

{/usr/local/texlive/2016/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
(./build/paper/text.aux) ) 
Here is how much of TeX's memory you used:
 214 strings out of 493014
 2390 string characters out of 6133351
 53601 words of memory out of 5000000
 3838 multiletter control sequences out of 15000+600000
 5457 words of font info for 20 fonts, out of 8000000 for 9000
 1141 hyphenation exceptions out of 8191
 22i,6n,19p,123b,187s stack positions out of 5000i,500n,10000p,200000b,80000s
</usr/local/texlive/2016/texmf-dist/fonts/type1/publi
c/amsfonts/cm/cmbx12.pfb></usr/local/texlive/2016/texmf-dist/fonts/type1/public
/amsfonts/cm/cmr12.pfb>
Output written on build/paper/text.pdf (1 page, 18345 bytes).
PDF statistics:
 16 PDF objects out of 1000 (max. 8388607)
 10 compressed objects within 1 object stream
 0 named destinations out of 1000 (max. 500000)
 1 words of extra memory for PDF output out of 10000 (max. 10000000)

 </details>